src/app/home/home.states.js

```diff
@@ -14,13 +14,12 @@ angular.module('home')
       templateUrl: 'app/home/home.html',
       resolve: {
         authentication: authProvider.requireAuthenticatedUser,
-        roundReport: function (deliveryRoundService, $rootScope, indexService) {
+        roundReport: function (deliveryRoundService, userStateService) {
           var key = ''
           var roundInfo = []
-          return indexService.getUserStates()
-            .then(function (states) {
-              var defaultState = angular.isArray(states) && states.length > 0 ? states[0]._id : ''
-              var state = angular.isDefined($rootScope.selectedState) ? $rootScope.selectedState._id : defaultState
+          return userStateService.getUserSelectedState()
+            .then(function (state) {
+              state = state || userStateService.stateMap.selectedState
```

Comment:
Asynchronous codes run side by side. setUserSelectedState sets the user selected state asynchronously, so its possible that localForage doesnt have the user state stored completely before this part of the application gets called. I saw this happened some few times especially at home.states case immediately a user logs in. So, if you try to call  getUserSelectedState you wont get any state returned yet. so what i did was, since i already knew that a stateMap.selectedState variable will be already set, so i get from it.The callback for localForage.getItem was called before the callback for localForage.setItem. and both of them are from different parts of the application, so i couldn't sequence them. UserStateService, and home.states